### PR TITLE
NativeAOT-LLVM: Merge oct 22

### PR DESCRIPTION
**Pull request by yowl**
I had problems in CI with the tests for the release builds for Win/Linux as per the comment below, so I've removed the `libs.tests` from the Release configuration.   Not great, but at least the NAOT smoke tests are running.

Otherwise I 

- added `RETAIL_CONFIG_VALUE_WITH_DEFAULT` so I can default `gcConservative` for `TARGET_WASM`
- excluded `DFEATURE_MANUALLY_MANAGED_CARD_BUNDLES` for Wasm
- excluded `__builtin___clear_cache` for Wasm (instruction cache I think)
- Added nop implementation in `ForEachPossibleObjectRef` in https://github.com/dotnet/runtimelab/pull/2219/commits/ef7c2158311eb4c3abbd223542e35472f685a424#diff-5c477abc72ab0f1d832dc5d3ce34a8467c2e41b805b509c567a0ef9e6d140cf2

The rest is mostly just moving stuff around.

### CODE REVIEW

**Comment by yowl** [comment]
Win x64 checked is building the tests ok locally, will try a rerun to see how consistent are the failures.

**Comment by yowl** [comment]
Win Release is failing writing to PDB 
```
CSC : error CS2012: Cannot open 'D:\a\_work\1\s\artifacts\obj\Microsoft.Win32.Registry\Release\net7.0-windows\Microsoft.Win32.Registry.pdb' for writing -- 'The requested operation cannot be performed on a file with a user-mapped section open.
```

Linux Release is failting with 
```
2023-03-29T14:28:41.9836184Z    ---> System.BadImageFormatException: Read out of bounds.
2023-03-29T14:28:41.9836424Z      at System.Reflection.Throw.OutOfBounds()
2023-03-29T14:28:41.9836675Z      at System.Reflection.Internal.MemoryBlock.PeekCompressedInteger(Int32, Int32& )
2023-03-29T14:28:41.9921723Z      at System.Reflection.Metadata.Ecma335.BlobHeap.GetMemoryBlock(BlobHandle)
2023-03-29T14:28:41.9922962Z      at System.Reflection.Metadata.MethodDebugInformation.GetSequencePoints()
2023-03-29T14:28:41.9924216Z      at Internal.TypeSystem.Ecma.PortablePdbSymbolReader.GetSequencePointsForMethod(Int32)+MoveNext() in /_/src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/PortablePdbSymbolReader.cs:line 126
2023-03-29T14:28:41.9925558Z      at ILCompiler.FeatureSwitchManager.GetMethodILWithInlinedSubstitutions(MethodIL) in /_/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/FeatureSwitchManager.cs:line 464
```

Will remove them from CI.

**Comment by yowl** [comment]
cc @dotnet/nativeaot-llvm

**Comment by yowl** [comment]
Going to put back to draft while try removing libs.test

File: src/coreclr/jit/importercalls.cpp [hunk 1]
```diff
@@ -4194,7 +4204,12 @@ void Compiler::impPopCallArgs(CORINFO_SIG_INFO* sig, GenTreeCall* call)
         }
         else
         {
+            // TODO-LLVM: lowerCallToShadowStack fails without the SignatureCorInfoType set (which is TARGET_WASM specific in NewCallArg 
+#if defined(TARGET_WASM)
+            arg = NewCallArg::Primitive(argNode, params[i - 1].CorType);
+#else
             arg = NewCallArg::Primitive(argNode, jitSigType);
+#endif
```
Comment (SingleAccretion) [inline src/coreclr/jit/importercalls.cpp]:
```suggestion
            arg = NewCallArg::Primitive(argNode, params[i - 1].CorType);
```
Or did that not work?

File: src/coreclr/jit/importercalls.cpp [hunk 2]
```diff
@@ -3,6 +3,12 @@
 
 #include "jitpch.h"
 
+#ifdef TARGET_WASM
+#include "llvm.h"
+#define max(a, b) (((a) > (b)) ? (a) : (b))
+#define min(a, b) (((a) < (b)) ? (a) : (b))
+#endif // TARGET_WASM
```
Comment (SingleAccretion) [inline src/coreclr/jit/importercalls.cpp:10]: This block can now be deleted from `importer.cpp`.

File: eng/pipelines/runtimelab.yml [hunk 3]
```diff
@@ -43,9 +43,9 @@ resources:
   containers:
   # Define auxilary containers used by the installer jobs
   - container: debpkg
-    image: mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-debpkg-20220504035737-cfdd435
+    image: mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-debpkg
   - container: rpmpkg
-    image: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-8-rpmpkg-20210714125410-daa5116
+    image: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-rpmpkg
```
Comment (SingleAccretion) [inline eng/pipelines/runtimelab.yml]: Particular reason to use different images?

Comment (yowl) [reply eng/pipelines/runtimelab.yml]: Not really, I just took what was in runtime.  I'll revert it.

File: src/coreclr/jit/importercalls.cpp (hunk 2 above)
Comment (yowl) [reply src/coreclr/jit/importercalls.cpp:10]: Thanks, removed

File: src/coreclr/jit/importercalls.cpp (hunk 1 above)
Comment (yowl) [reply src/coreclr/jit/importercalls.cpp]: I didn't try it to be honest, but yes that does work and is now changed as per the suggestion, thanks.

**Review by SingleAccretion — APPROVED** [review]
:shipit:

**Comment by jkotas** [comment]
Thank you!

**Pull request merged**